# Statistical Conclusion And Claim Boundary

The experiment answers a narrow but useful question: can the pipeline recover a known RM/PM demand process when complete production-plan and BOM signals exist?


In [1]:
from pathlib import Path
import json
import pandas as pd
try:
    from IPython.display import Image, display
except ImportError:
    class Image:
        def __init__(self, filename): self.filename = filename
        def __repr__(self): return f"Image({self.filename})"
    def display(value): print(value)

ROOT = Path.cwd()
if ROOT.name != "v8_controlled_synthetic_validation":
    candidate = ROOT / "Ai miroservices/modeling/v8_controlled_synthetic_validation"
    ROOT = candidate if candidate.exists() else ROOT
OUT = ROOT / "outputs"
DATA = OUT / "data"
PLOTS = OUT / "plots"
EVAL = OUT / "evaluator"
summary = json.loads((OUT / "run_summary.json").read_text())


In [2]:
summary


{'seed': 20260711,
 'data_tier': 'CONTROLLED_SYNTHETIC_GROUND_TRUTH',
 'materials': 120,
 'raw_materials': 90,
 'packaging_materials': 30,
 'finished_goods': 24,
 'bom_rows': 211,
 'bom_coverage_controlled_pct': 100.0,
 'history_months': 72,
 'tuning_months': 6,
 'selection_months': 6,
 'test_months': 12,
 'best_lightgbm_params': {'num_leaves': 31,
  'learning_rate': 0.05,
  'min_child_samples': 30,
  'reg_lambda': 0.5,
  'n_estimators': 320},
 'locked_champion': 'extra_trees_causal',
 'selection_champion_metrics': {'model': 'extra_trees_causal',
  'rows': 720,
  'materials': 120,
  'WAPE': 0.0810995644250875,
  'MAE': 698.337999045157,
  'RMSE': 1456.85484423038,
  'Bias': -0.009952376486637864,
  'under_forecast_rate': 0.4722222222222222,
  'shock_WAPE': 0.09176384160838098},
 'test_champion_metrics': {'model': 'extra_trees_causal',
  'rows': 1440,
  'materials': 120,
  'WAPE': 0.08335697840628727,
  'MAE': 736.7584634403185,
  'RMSE': 1408.4608905558036,
  'Bias': -0.001509353087020

In [3]:
evaluator_summary=json.loads((EVAL/'evaluator_run_summary.json').read_text())
display(pd.DataFrame([evaluator_summary]).T)


,0
data_tier,CONTROLLED_SYNTHETIC_GROUND_TRUTH
purpose,Independent v8 replication of the shared neura...
history_months,24
horizon_months,12
seeds,"[17, 42, 101, 303, 707]"
production_decision_eligible,False
external_population_validity,UNVERIFIED
leaderboard,"[{'split': 'selection', 'model_name': 'CONV1D_..."


## Supported Conclusions

- The pipeline successfully recovers the controlled causal structure.
- The selection-locked Extra Trees causal model reaches approximately 8.34% WAPE on twelve untouched origins.
- Random Forest is statistically indistinguishable from the locked Extra Trees candidate in this test, showing that causal tree ensembles recover similar behavior.
- Direct-history LightGBM is significantly worse than BOM-plan forecasting, demonstrating the value of known production/BOM inputs.
- Conformal intervals achieve approximately their nominal 90% coverage.
- Heavy tails and heteroscedasticity remain even when point forecasts are good; Gaussian residual claims remain invalid.

## Unsupported Conclusions

- Synthetic success does not prove real production accuracy.
- Synthetic success does not prove that every v7 error is exclusively a data problem.
- A complete synthetic BOM cannot be represented as a real validated Hemas BOM.
- Hyperparameter search cannot replace real material-issue and production-order history.

## Operational Decision

Keep v8 as the controlled validation harness and explicit project-operational simulation seed. Its rows may drive the integrated demonstration only while their synthetic provenance and decision scope remain visible. They are not evidence of external production accuracy.
